lib/posix/signal.c

```diff
@@ -100,3 +100,8 @@ char *strsignal(int signum)
 
 	return buf;
 }
+
+int sigprocmask(int how, const sigset_t *ZRESTRICT set, sigset_t *ZRESTRICT oset)
+{
+	return pthread_sigmask(how, set, oset);
```

Comment:
From the spec

> The use of the sigprocmask() function is unspecified in a multi-threaded process.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_sigmask.html

What's tricky is that
- Zephyr does not currently support processes
- Zephyr does not support querying how many threads are active at any given time in a single process

```suggestion
	if (!IS_ENABLED(CONFIG_MULTITHREADING)) {
		return pthread_sigmask(how, set, oset);
	}

	__ASSERT(false, "In multi-threaded environments, please use pthread_sigmask() instead of sigprocmask()");
	
	errno = ENOSYS;
	return -1;
```